File: Java/Level1/성격 유형 검사하기.java
import java.util.HashMap;

class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        int N = survey.length;
        HashMap<Character, Integer> score = new HashMap<>();
        char[][] types = {{'R', 'T'}, {'C', 'F'}, {'J', 'M'}, {'A', 'N'}};

        for (char[] type : types) {
            score.put(type[0], 0);
            score.put(type[1], 0);
        }

        for (int i = 0; i < N; i++) {
            int c = choices[i];
            if (c < 4)
                score.put(survey[i].charAt(0), score.get(survey[i].charAt(0)) + (4 - c));
            else
                score.put(survey[i].charAt(1), score.get(survey[i].charAt(1)) + (c - 4));
        }

        for (char[] type : types) {
            char t1 = type[0], t2 = type[1];
            answer += (score.get(t1) >= score.get(t2)) ? t1 : t2;
        }

        return answer;
    }
}

File: programmers/C++/Level1/성격 유형 검사하기.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> survey, vector<int> choices) {
    string answer = "";
    int N = survey.size();
    unordered_map<char, int> score;

    for (int i = 0; i < N; i++) {
        int c = choices[i];
        if (c < 4)
            score[survey[i][0]] += (4 - c);
        else
            score[survey[i][1]] += (c - 4);
    }

    string types[4] = { "RT", "CF", "JM", "AN" };
    for (string t : types) {
        char t1 = t[0], t2 = t[1];
        answer += (score[t1] >= score[t2]) ? t1 : t2;
    }

    return answer;
}